### ISSUE TITLE
loss = 0 when sft qwen3.5-9B

### ISSUE THREAD

**Issue by shun-qi** [modelscope/ms-swift#9499]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

除了第一条log情况，后面的loss全为0
2026-06-04 21:41:29.421
{'loss': '0', 'grad_norm': '1.414', 'learning_rate': '2.67e-06', 'token_acc': '4.23e-06', 'epoch': '0.02737', 'global_step/max_steps': '4/294', 'elapsed_time': '28m 29s', 'remaining_time': '1d 10h 25m 8s', 'memory(GiB)': '44.59', 'train_speed(s/it)': '427.3'}
2026-06-04 21:35:21.137		
{'loss': '0', 'grad_norm': '1.414', 'learning_rate': '2e-06', 'token_acc': '3.98e-06', 'epoch': '0.02053', 'global_step/max_steps': '3/294', 'elapsed_time': '22m 21s', 'remaining_time': '1d 12h 7m 36s', 'memory(GiB)': '44.59', 'train_speed(s/it)': '446.9'}
2026-06-04 21:29:07.052	
{'loss': '0', 'grad_norm': '1.414', 'learning_rate': '1.33e-06', 'token_acc': '4.41e-06', 'epoch': '0.01369', 'global_step/max_steps': '2/294', 'elapsed_time': '16m 7s', 'remaining_time': '1d 15h 12m 19s', 'memory(GiB)': '44.59', 'train_speed(s/it)': '483.4'}
2026-06-04 21:22:57.641	
{'loss': '12.9', 'grad_norm': '9.317', 'learning_rate': '6.7e-07', 'token_acc': '9.1e-06', 'epoch': '0.006843', 'global_step/max_steps': '1/294', 'elapsed_time': '9m 57s', 'remaining_time': '2d 0h 36m 48s', 'memory(GiB)': '28.14', 'train_speed(s/it)': '597.3'}
2026-06-04 21:13:03.628	
[INFO:swift] use_logits_to_keep: False

### How to Reproduce / 如何复现

# === 通用运行环境 ===
export NCCL_IB_DISABLE=1
export NCCL_P2P_DISABLE=1
export OMP_NUM_THREADS=4
export PYTORCH_CUDA_ALLOC_CONF='expandable_segments:True'

# rank / world_size ===
# NODE_IP_LIST 形如 "1.2.3.4:8,5.6.7.8:8,..."
# 拆分出节点 IP 列表
NODE_IPS=$(echo "$NODE_IP_LIST" | sed 's/:[^,]*//g' | tr ',' '\n')
WORLD_SIZE=$(echo "$NODE_IPS" | wc -l)

# 找出当前节点在列表中的位置（从 0 开始）
GLOBAL_RANK=$(echo "$NODE_IPS" | grep -nx "$LOCAL_IP" | head -n1 | cut -d: -f1)
GLOBAL_RANK=$((GLOBAL_RANK - 1))

# === 自动检测本机 GPU 数 ===
if command -v nvidia-smi >/dev/null 2>&1; then
  NPROC_PER_NODE=$(nvidia-smi -L | wc -l)
else
  NPROC_PER_NODE=1
fi
if [ -z "$NPROC_PER_NODE" ] || [ "$NPROC_PER_NODE" -le 0 ]; then
  NPROC_PER_NODE=1
fi

# 主节点取 IP 列表中的第一个
MASTER_ADDR=$(echo "$NODE_IPS" | head -n1)
MASTER_PORT=29500

# 构造 CUDA_VISIBLE_DEVICES = 0,1,...,NPROC_PER_NODE-1
CUDA_VISIBLE_DEVICES=$(seq -s, 0 $((NPROC_PER_NODE - 1)))

export NNODES=$WORLD_SIZE
export NPROC_PER_NODE
export NODE_RANK=$GLOBAL_RANK
export MASTER_ADDR
export MASTER_PORT
export CUDA_VISIBLE_DEVICES

echo "[start.sh] LOCAL_IP=${LOCAL_IP}"
echo "[start.sh] NODE_IP_LIST=${NODE_IP_LIST}"
echo "[start.sh] GLOBAL_RANK=${GLOBAL_RANK}  WORLD_SIZE=${WORLD_SIZE}"
echo "[start.sh] NNODES=${NNODES}  NPROC_PER_NODE=${NPROC_PER_NODE}  NODE_RANK=${NODE_RANK}"
echo "[start.sh] MASTER_ADDR=${MASTER_ADDR}  MASTER_PORT=${MASTER_PORT}"
echo "[start.sh] CUDA_VISIBLE_DEVICES=${CUDA_VISIBLE_DEVICES}"

swift sft \
    --model Qwen3.5-9B \
    --dataset "sample_1w_12_4_0604_clean.jsonl" \
    --split_dataset_ratio 0.01 \
    --packing false \
    --tuner_type full \
    --num_train_epochs 2 \
    --save_steps 200 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-5 \
    --freeze_vit false \
    --freeze_aligner false \
    --gradient_accumulation_steps $(expr 256 / \( $NNODES \* $NPROC_PER_NODE \)) \
    --eval_steps 20 \
    --save_strategy steps \
    --save_total_limit 1 \
    --logging_steps 1 \
    --lr_scheduler_type cosine \
    --max_length 10000 \
    --output_dir "${OUTPUT_DIR}" \
    --report_to tensorboard \
    --logging_dir "$EVENTS_PATH" \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 8 \
    --dataset_num_proc 2 \
    --use_liger_kernel true \
    --deepspeed zero3 \
    --torch_dtype bfloat16 \
    --attn_impl flash_attn \
    --system "system.txt"

### Additional Information / 补充信息

训练数据集是平台提供的sample脚本sample得到的包含messages，images，rejected_response
